File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int checker(int x) {
    if (x < 2) {
        return 6;
    }
    else
        return 7 - x;
}

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int lapf = 0;
    int laps = 0;
    for (int i : lottos) {
        if (i == 0) {
            laps++;
        }
        for (int j : win_nums) {
            if (i == j) {
                lapf++;
            }
        }
    }
    answer.push_back(checker(lapf+laps));
    answer.push_back(checker(lapf));
    return answer;
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.kt
class Solution {
    private fun getCheck(n: Int):Int=if(n<2)6 else 7-n

    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        val equals= mutableListOf(0,0)
        lottos.forEach { i->
            if(i==0)equals[0]++
            win_nums.forEach { j->
                if(i==j)equals[1]++
            }
        }
        return intArrayOf(getCheck(equals[1]+equals[0]), getCheck(equals[1]))
    }
}